test/role.test.js

```diff
@@ -208,4 +210,64 @@ describe('role model', function() {
 
   });
 
+  describe('listByPrincipalType', function() {
+    var sandbox;
+
+    beforeEach(function() {
+      sandbox = sinon.sandbox.create();
+    });
+
+    afterEach(function() {
+      sandbox.restore();
+    });
+
+    it('should fetch all models assigned to the role', function(done) {
+      var principalTypesToModels = {};
+      var runs = 0;
+      var mappings;
+
+      principalTypesToModels[RoleMapping.USER] = User;
+      principalTypesToModels[RoleMapping.APPLICATION] = Application;
+      principalTypesToModels[RoleMapping.ROLE] = Role;
+
+      mappings = Object.keys(principalTypesToModels);
+
+      mappings.forEach(function(principalType) {
+        var Model = principalTypesToModels[principalType];
+        Model.create({name:'test', email:'x@y.com', password: 'foobar'}, function(err, model) {
+          Role.create({name:'testRole'}, function(err, role) {
+            role.principals.create({principalType: principalType, principalId: model.id}, function(err, p) {
+              var pluralName = Model.pluralModelName.toLowerCase();
+              role[pluralName](function(err, models) {
+                assert(!err);
+                assert.equal(models.length, 1);
+                if (++runs === mappings.length) {
+                  done();
+                }
+              });
+            });
+          });
+        });
+      });
+    });
+
+    it('should apply query', function(done) {
+      User.create({name: 'Raymond', email: 'x@y.com', password: 'foobar'}, function(err, user) {
+        Role.create({name: 'userRole'}, function(err, role) {
+          role.principals.create({principalType: RoleMapping.USER, principalId: user.id}, function(err, p) {
+            var query = {fields:['id', 'name']};
+            sandbox.spy(User, 'find');
```

Comment:
> A spy is used to examine the calls of a function and the arguments passed to it.

Haha yes I know this part. :)

> In unit tests, its important to test with laser focus.

Agreed... And we aren't doing a good job of this really. But I prefer the same kind of focus you are talking about with PRs so we can avoid having to go off track in the PR comments like this :(.

> I would suggest an iterative approach to adding sinon so that some tests can benefit while others are being converted.

Sounds good... just not alongside another PR. Please move it to a separate one. That way we can get some more peoples (eg. @superkhau @bajtos) opinions and chat about it.

>  I could set it up in a way that makes it easy to use in other tests, and implement in some of the other tests when I have time. I saw a few other places that could benefit from it.

Since you might not have time, even just starting a conversation about the specifics could be helpdul. Just keep in mind that this isn't really a priority.